libs/s3/s3_tagging_test_lib.py

```diff
@@ -471,3 +473,44 @@ def get_object_with_tagging(
             raise CTException(err.S3_CLIENT_ERROR, error.args[0])
 
         return True, response
+
+    def set_encoded_tag_values(
+            self,
+            bucket_name: str = None,
+            encoding_type: str = "utf-8") -> tuple:
+        """
+        Set tag to a bucket with encoded tag values.
+
+        :param bucket_name: Name of bucket.
+        :param encoding_type: encoding type e.g. base64,utf-8
+        :return: True or False, response and encoded tag set.
+        """
+        try:
+            LOGGER.info("Set bucket tag with encoded key value pair.")
+            key = ''.join((secrets.choice(string.printable) for i in range(8)))
+            value = ''.join((secrets.choice(string.printable) for i in range(8)))
+            tag_set = list()
+            if encoding_type == 'utf-8':
+                key_encode = key.encode('utf-8')
+                value_encode = value.encode('utf-8')
+            elif encoding_type == 'base64':
+                key_encode = base64.b64encode(key.encode())
+                value_encode = base64.b64encode(value.encode())
+            else:
```

Comment:
Can you precheck the b64 and utf-8 at the start and raise a precise EncodingNotSupported exception.